sky/provision/kubernetes/utils.py

```diff
@@ -185,6 +185,48 @@ def get_accelerator_from_label_value(cls, value: str) -> str:
                 f'Invalid accelerator name in GKE cluster: {value}')
 
 
+class GFDLabelFormatter(GPULabelFormatter):
+    """GPU Feature Discovery label formatter
+
+    NVIDIA GPUs nodes are labeled by GPU feature discovery
+    e.g. nvidia.com/gpu.product=NVIDIA-H100-80GB-HBM3
+    https://github.com/NVIDIA/gpu-feature-discovery
+
+    GPU feature discovery is included as part of the
+    NVIDIA GPU Operator:
+    https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/overview.html
+    """
+
+    LABEL_KEY = 'nvidia.com/gpu.product'
+
+    @classmethod
+    def get_label_key(cls) -> str:
+        return cls.LABEL_KEY
+
+    @classmethod
+    def get_accelerator_from_label_value(cls, value: str) -> str:
```

Comment:
Do we also need to implement `get_label_value`?

Reply:
I don't think we need to. It's also tricky to create since the set of accelerator types can have one -> many mappings to GFD label values. One that I've run into is PCIE vs SXM variants of the A100 and H100 gpus. This is the main reason I [changed the logic](https://github.com/asaiacai/skypilot/blob/031921564bf51ce8920f23ea81c912328a3b8726/sky/provision/kubernetes/utils.py#L484) to check by going from (acc_type --> label value) to (label value --> acc_type). Outside of there, `get_label_value` is [only used for autoscaling](https://github.com/asaiacai/skypilot/blob/031921564bf51ce8920f23ea81c912328a3b8726/sky/provision/kubernetes/utils.py#L436) which GFDLabeler wouldn't be used.

Reply:
The one->many problem makes sense. Should we add something like this to make a note of this:
```
    @classmethod
    def get_label_value(cls, accelerator: str) -> str:
        """An accelerator can map to many Nvidia GFD labels (e.g., <example>). As a result, we do not support get_label_value for GFDLabelFormatter."""
        raise NotImplementedError
```

We should also add to the class GFDLabelFormatter docstr something like:
```
This LabelFormatter cannot be used in autoscaling clusters since ...
```